src/main/java/com/fundynamic/d2tm/game/math/Vector2D.java

```diff
@@ -1,35 +1,46 @@
 package com.fundynamic.d2tm.game.math;
 
-public class Vector2D {
+public class Vector2D<T extends Number> {
 
-    private final int x, y;
+    private final T x, y;
 
-    public Vector2D(int x, int y) {
+    public static Vector2D zero() {
+        return new Vector2D(0, 0);
+    }
+
+    public Vector2D(T x, T y) {
         this.x = x;
         this.y = y;
     }
 
-    public int getX() {
+    public T getX() {
         return x;
     }
 
-    public int getY() {
+    public T getY() {
         return y;
     }
 
-    public Vector2D moveUp() {
-        return new Vector2D(x, y - 1);
+    public Vector2D<Float> move(float xVelocity, float yVelocity, float speed) {
```

Comment:
because it extends from `Number` it is not able to tell what you're adding/substracting, unless you make it explicit (int/float, etc). And since we're using `floatValue` we make it explicit and should return that. Movement is float operations anyway. (more info: http://stackoverflow.com/questions/2721390/how-to-add-two-java-lang-numbers)